webui/react/src/components/NotebookModal.tsx

```diff
@@ -0,0 +1,235 @@
+  const [ showFullConfig, setShowFullConfig ] = useState(false);
+  const [ templates, setTemplates ] = useState<Template[]>([]);
+  const [ resourcePools, setResourcePools ] = useState<ResourcePool[]>([]);
+  const [ showResourceType, setShowResourceType ] = useState(false);
+  const [ resourceType, setResourceType ] = useState<ResourceType | undefined>
+  (storage.getWithDefault(STORAGE_KEY, { type: undefined }).type);
+  const [ form ] = Form.useForm();
+
+  const fetchTemplates = useCallback(async () => {
+    try {
+      setTemplates(await getTaskTemplates({}));
+    } catch {}
+  },[]);
+
+  useEffect(() => {
+    fetchTemplates();
+  }, [ fetchTemplates ]);
+
+  const fetchResourcePools = useCallback(async () => {
+    try {
+      setResourcePools( await getResourcePools({}));
+    } catch {}
+  }, []);
+
+  useEffect(() => {
+    fetchResourcePools();
+  }, [ fetchResourcePools ]);
+
+  const fetchConfig = useCallback(async () => {
+    try {
+      const values: NotebookConfig = form.getFieldsValue(true);
+      const config = await previewNotebook(
+        values.type === ResourceType.CPU ? 0 : values.slots,
+        values.template === 'default'? undefined : values.template,
+        values.name,
+        values.pool,
+      );
+      form.setFieldsValue({ config: yaml.dump(config) });
+    } catch {}
+  }, [ form ]);
+
+  useEffect(()=> {
+    if (showFullConfig) fetchConfig();
+  }, [ showFullConfig, fetchConfig ]);
+
+  const storeConfig = useCallback((_, values: NotebookConfig) => {
+    delete values.name;
+    storage.set(STORAGE_KEY,values);
+  }, [ storage ]);
+
+  const handleSecondary = useCallback(async () => {
+    if (showFullConfig) {
+      setShowFullConfig(false);
+    } else {
+      try {
+        await form.validateFields();
+        setShowFullConfig(true);
+      } catch (e) {}
+    }
+  },[ form, showFullConfig ]);
+
+  const handleCreateEnvironment = useCallback(
+    (values) =>{
+      if (showFullConfig) {
+        launchNotebook(yaml.load(form.getFieldValue('config')) as RawJson);
+      } else {
+        launchNotebook(
+          undefined,
+          resourceType === ResourceType.CPU ? 0 : values.slots,
+          values.template === 'default' ? undefined : values.template,
+          values.name,
+          values.pool,
+        );
+      }
+      if (onLaunch) onLaunch();
+    },
+    [ showFullConfig, form, onLaunch, resourceType ],
+  );
+
+  const handleResourcePoolUpdate = useCallback((selectedPoolName: string) => {
+    const selectedPool = resourcePools.find(pool => pool.name === selectedPoolName);
+    if (!selectedPool) return;
+    const hasCPUCapacity = selectedPool.cpuContainerCapacityPerAgent > 0;
+    const hasGPUCapacity = selectedPool.slotsPerAgent && selectedPool.slotsPerAgent > 0;
+    if (hasCPUCapacity && hasGPUCapacity) {
+      setShowResourceType(true);
+    } else if (hasCPUCapacity) {
+      setResourceType(ResourceType.CPU);
+      setShowResourceType(false);
```

Comment:
If there is CPU capacity, the `resources.slots` should be filled with 0 rather than 1.

Reply:
If there is only CPU capacity, the slots field doesn't show up and it is treated as 0 slots.